src/ast/query.rs

```diff
@@ -665,6 +665,8 @@ pub enum TableFactor {
         /// Optional version qualifier to facilitate table time-travel, as
         /// supported by BigQuery and MSSQL.
         version: Option<TableVersion>,
+        /// Partition selection, supported by MySQL.
+        partitions: Vec<Expr>,
```

Comment:
I wonder why this is an `Expr` ? It seems like each element is a `partition_name` 

https://dev.mysql.com/doc/refman/8.0/en/create-table.html

So perhaps this should be something like 

```suggestion
        partitions: Vec<Ident>,
```

And then call https://docs.rs/sqlparser/latest/sqlparser/parser/struct.Parser.html#method.parse_identifier to parse them below?

Reply:
done